A28-xds-traffic-splitting-and-routing.md

```diff
@@ -0,0 +1,668 @@
+corresponding matching criteria. The RPCs will be matched against the
+criteria, and will be distributed to the matched child.
+
+### Child policies
+
+The child policies will be one of
+
+*   `cds_experimental`
+*   `weighted_target_experimental`
+    *   Because this will only have CDS as children, this is essentially
+        `weighted_CDS`
+
+More child policies can be added in the future.
+
+### LB policy config
+
+The `xds_routing_experimental` LB policy will have the following
+configuration:
+
+```proto
+/*
+
+Route List                         Action map
++---------+--------+               +-------+-----------------+--------------+
+| matcher | action |               | name  | policy          | config       |
++------------------+               +----------------------------------------+
+| /s/m1   | wt_1   | +-----------> | wt_1  | weighted_target | <config_1>   |
++------------------+               +----------------------------------------+
+| /s/m2   | cds_1  | +----\ /----> | wt_2  | weighted_target | <config_2>   |
++------------------+       x       +----------------------------------------+
+| /s2/m3  | wt_2   | +----/ \----> | cds_1 | CDS             | <cds_config> |
++------------------+            -> +-------+-----------------+--------------+
+| /s2/m4  | cds_1  | +---------/
++---------+--------+
+
+ */
+
+message XdsRouting {
+  message Action {
+    repeated LoadBalancingConfig child_policy = 1;
+  }
+  // Route represents a routing rule to send an RPC. It contains a matcher and
+  // the name of the action to take. The action with the name can be found in
+  // the action list.
+  //
+  // Route contains the action name instead of embedding the Action, so Action
+  // can be shared and reused by multiple routes.
+  message Route {
+    message HeaderMatcher {
+      message Int64Range {
+        int64 start = 1;
+        int64 end = 2;
+      }
+      string name = 1;
+      oneof header_match_specifier {
+        string exact_match = 2;
+        string regex_match = 3;
+        Int64Range range_match = 4;
+        bool present_match = 5;
+        string prefix_match = 6;
+        string suffix_match = 7;
+      }
+      bool invert_match = 8;
+    }
+
+    oneof path_matcher {
+      string path = 1;
+      string prefix = 2;
+      string regex = 3;
+    }
+    repeated HeaderMatcher headers = 3;
+
+    // match_fraction specifies the percentage this route should be considered.
+    // For fraction `N/D`, route is considered if `random(0,D) <= N`.
+    //
+    // The number in this field specifies the numerator.  The implicit
+    // denominator is 1M (1,000,000). If the numerator specified is bigger than
+    // the denominator, the final fractional percentage is capped at 1 (100%).
+    //
+    // The default value (when this field is not set) is 100%, which means
+    // always consider this route (continue to evaluate the remaining match
+    // criteria).
+    google.protobuf.UInt32Value match_fraction = 4;
```

Comment:
Which field in xDS `RouteConfiguration` does this correspond with?

Reply:
It corresponds to the `runtime_fraction` field:

https://github.com/envoyproxy/envoy/blob/eeaebc1bea6d5b52368df18378797b1dcc163141/api/envoy/api/v2/route/route_components.proto#L459

As mentioned below, we will support only the `default_value` field, not the `runtime_key` field, since we don't have the concept of runtime.

Reply:
I see, maybe explain this explicitly in the comment or writeup, I don't see many mentions of this field when searching.

Reply:
Added one line in section "Service config generating">"Route list", line 456.